File: backend/SentinelOps.Application/Common/Models/AlertModels.cs
using SentinelOps.Domain.Enums;

namespace SentinelOps.Application.Common.Models;

public class AlertDto
{
    public Guid Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Message { get; set; } = string.Empty;
    public string Source { get; set; } = string.Empty;
    public AlertStatus Status { get; set; }
    public Severity Severity { get; set; }
    public double? MetricValue { get; set; }
    public double? Threshold { get; set; }
    public Guid? IncidentId { get; set; }
    public DateTime CreatedAt { get; set; }
}

public class CreateAlertDto
{
    public string Title { get; set; } = string.Empty;
    public string Message { get; set; } = string.Empty;
    public string Source { get; set; } = string.Empty;
    public Severity Severity { get; set; }
    public double? MetricValue { get; set; }
    public double? Threshold { get; set; }
}

public class DashboardStatsDto
{
    public int TotalIncidents { get; set; }
    public int OpenIncidents { get; set; }
    public int CriticalIncidents { get; set; }
    public int ResolvedToday { get; set; }
    public int ActiveAlerts { get; set; }
    public int SlaBreaches { get; set; }
    public double AverageResolutionHours { get; set; }
}

File: frontend/sentinelops-ui/src/app/shared/models/alert.models.ts
import { Severity } from './incident.models';

export enum AlertStatus {
  Active = 0,
  Acknowledged = 1,
  Resolved = 2,
  Suppressed = 3
}

export interface AlertDto {
  id: string;
  title: string;
  message: string;
  source: string;
  status: AlertStatus;
  severity: Severity;
  metricValue: number | null;
  threshold: number | null;
  incidentId: string | null;
  createdAt: Date;
}

export interface DashboardStatsDto {
  totalIncidents: number;
  openIncidents: number;
  criticalIncidents: number;
  resolvedToday: number;
  activeAlerts: number;
  slaBreaches: number;
  averageResolutionHours: number;
}
